### PR TITLE
Send repeatedly sync requests via gossip

### LINKED ISSUE #1011: Gossip periodcally syncs state
Node sends every X seconds an sync request to another member. This will be then merged with the current state.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
Gossip nodes are sending repeatedly sync requests in round-robin fashion to other members.
Sync interval is per default 15 seconds, but can be configured.

closes #1011

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 86d50957a89f7e89f8c4dee09b9771632c18ffe4
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, ci**

Guidelines are available at https://github.com/zeebe-io/zeebe/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: gossip/src/main/java/io/zeebe/gossip/failuredetection/SyncController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import io.zeebe.transport.ClientResponse;
+import io.zeebe.util.sched.ActorControl;
+import io.zeebe.util.sched.future.ActorFuture;
+import java.time.Duration;
+import org.agrona.DirectBuffer;
+import org.slf4j.Logger;
+
+public class SyncController {
+  private static final Logger LOG = Loggers.GOSSIP_LOGGER;
+  private static final Duration SYNC_INTERVAL = Duration.ofSeconds(15);
```
Comment (menski) [inline gossip/src/main/java/io/zeebe/gossip/failuredetection/SyncController.java]: remove this constant and use the configuration value instead

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java [hunk 2]
```diff
@@ -170,8 +170,10 @@ public void onEvent(SocketAddress sender, DirectBuffer payload) {
             readSocketAddress(offset, payloadCopy, replicationApi);
 
             final NodeInfo newMember = new NodeInfo(clientApi, managementApi, replicationApi);
-            topology.addMember(newMember);
-            notifyMemberAdded(newMember);
+            final boolean memberAdded = topology.addMember(newMember);
+            if (memberAdded) {
```
Comment (menski) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java]: just for better understanding, why was this change needed?

**Review by menski — CHANGES_REQUESTED** [review]

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java (hunk 2 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java]: I thought that it make no sense to notify the listeners if the member already exist, but maybe this can not happen.

File: gossip/src/main/java/io/zeebe/gossip/failuredetection/SyncController.java (hunk 1 above)
Comment (Zelldon) [reply gossip/src/main/java/io/zeebe/gossip/failuredetection/SyncController.java]: yo sorry forgot this

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java (hunk 2 above)
Comment (menski) [reply broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java]: it's fine, was sure if this is now a requirement that it never happens twice, but if it's just a optimization than it's fine

**Review by menski — APPROVED** [review]

**Comment by menski** [comment]
bors retry

**Comment by Zelldon** [comment]
bors retry

**Comment by menski** [comment]
Ok @Zelldon I think you where right the "branch has to be up to date with develop" also broke the bot. Let's try without it.

bors retry

**Comment by menski** [comment]
bors retry
